GEOparse/GEOTypes.py

```diff
@@ -609,50 +616,36 @@ def download_SRA(self, email, directory='./', **kwargs):
                     # the directory name cannot contain many of the signs
                     re.sub(name_regex, '_', self.metadata['title'][0]))))
 
-            utils.mkdir_p(os.path.abspath(directory_path))
+        if nproc==1 and not return_list:
+            # No need to parallelize, simple run for all sra:
+            downloaded_paths = []
             for path in df['download_path']:
-                sra_run = path.split("/")[-1]
-                logger.info("Analysing %s" % sra_run)
-                url = ftpaddres.format(range_subdir=sra_run[:6],
-                                       file_dir=sra_run)
-                logger.debug("URL: %s", url)
-                filepath = os.path.abspath(
-                    os.path.join(directory_path, "%s.sra" % sra_run))
-                utils.download_from_url(url, filepath, aspera=aspera)
-
-                if filetype in ["fasta", "fastq"]:
-                    if utils.which('fastq-dump') is None:
-                        raise NoSRAToolkitException(
-                            "fastq-dump command not found")
-                    ftype = ""
-                    if filetype == "fasta":
-                        ftype = " --fasta "
-                    cmd = "fastq-dump"
-                    for fqoption, fqvalue in iteritems(fastq_dump_options):
-                        if fqvalue:
-                            cmd += (" --%s %s" % (fqoption, fqvalue))
-                        else:
-                            cmd += (" --%s" % fqoption)
-                    cmd += " %s --outdir %s %s"
-                    cmd = cmd % (ftype, directory_path, filepath)
-                    logger.debug(cmd)
-                    process = subprocess.Popen(cmd, stdout=subprocess.PIPE,
-                                               stderr=subprocess.PIPE,
-                                               shell=True)
-                    logger.info("Converting to %s/%s*.%s.gz\n" % (
-                        directory_path, sra_run, filetype))
-                    pout, perr = process.communicate()
-                    downloaded_paths += glob.glob(os.path.join(
-                        directory_path,
-                        "%s*.%s.gz" % (sra_run, filetype)
-                    ))
-                    if not keep_sra and filetype != 'sra':
-                        # Delete sra file
-                        os.unlink(filepath)
-                    else:
-                        downloaded_paths.append(filepath)
+                filepath = utils.download_unpack_SRA(path, ftpaddres,
```

Comment:
I would try to keep all function parameters in one line or all in separate lines. In some case I see I am not following this but this is also my mistake.